orttraining/orttraining/core/graph/pipeline_transformer.cc

```diff
@@ -13,11 +13,59 @@ using namespace onnxruntime::graph_utils;
 namespace onnxruntime {
 namespace training {
 
-void GetPipelineSendOutput(const Graph& graph, std::string& loss_name) {
+void CreateFakeOutput(
```

Comment:
For DeepSpeed, I think graph-level outputs are broadcasted to all stages. For FairScale, the user has to use if-else to avoid accessing outputs at intermediate stages.

---
In reply to: [527637087](https://github.com/microsoft/onnxruntime/pull/5815#discussion_r527637087) [](ancestors = 527637087)